test/unit_tests/test_ops.py

```diff
@@ -83,8 +96,17 @@ def _assert_decomposition(pl_op, params):
     ]
     index_substitutions = {i: i + num_wires for i in range(num_wires)}
     next_index = num_indices
+
+    # get decomposed gate
+    if params:  # for parametrized gates
+        decomposed_op = pl_op.decomposition(*params, wires=wires)
+        pl_op_matrix = pl_op._matrix(*params)
+    else:  # for non parametrized gates
+        decomposed_op = pl_op.decomposition(wires=wires)
+        pl_op_matrix = pl_op._matrix()
```

Comment:
Can we use comprehension here?

Reply:
Do you meaning changing something like below?
`pl_op_matrix = pl_op._matrix(*params) if params else pl_op._matrix()`

Reply:
Yes, this would avoid duplication of variable assignment and improve readability overall, IMO. What do you think? 

Check if something like this works: 
```
decomposed_op, pl_op_matrix = (pl_op.decomposition(*params, wires=wires), pl_op._matrix(*params)) if params else (pl_op.decomposition(wires=wires), pl_op._matrix())
```